src/runtime.rs

```diff
@@ -0,0 +1,14 @@
+use std::time::Duration;
+
+pub trait Runtime {
+    fn sleep(duration: Duration) -> impl std::future::Future<Output = ()> + Send;
+}
```

Comment:
I feel that `Runtime` is too broad as we are only looking for an sleep function implementation, it's probably fine using `Sleeper` instead, here and in the `Builder` struct.